# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: mozglue/baseprofiler/core/ProfileBuffer.h
# Commit: fd3a29881783
# Full Hash: fd3a29881783aad22258c96c6ea04220a40f2648
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2019-09-18 09:56:40
# Regressor Bug: 1576551
# File Overlap Count: 1
# Description:
#   Bug 1576551 - Store BlocksRingBuffer outside of ProfileBuffer - r=gregtatum
#   
#   The main `BlocksRingBuffer`s will be stored in `CorePS` (outside of
#   `ProfileBuffer`s), as we need to be able to safely access the underlying buffers
#   when profilers are not enabled.
# ==============================================================================

diff -r 3dfc76ea7bd9 -r fd3a29881783 mozglue/baseprofiler/core/ProfileBuffer.h
--- a/mozglue/baseprofiler/core/ProfileBuffer.h	Wed Sep 18 01:19:14 2019 +0000
+++ b/mozglue/baseprofiler/core/ProfileBuffer.h	Wed Sep 18 01:19:21 2019 +0000
@@ -30,8 +30,9 @@
   using BlockIndex = BlocksRingBuffer::BlockIndex;
 
   // ProfileBuffer constructor
-  // @param aCapacity The capacity of the buffer.
-  explicit ProfileBuffer(PowerOfTwo32 aCapacity);
+  // @param aBuffer The BlocksRingBuffer to use as buffer manager.
+  // @param aCapacity The capacity of the buffer in memory.
+  ProfileBuffer(BlocksRingBuffer& aBuffer, PowerOfTwo32 aCapacity);
 
   ~ProfileBuffer();
 
@@ -113,7 +114,7 @@
                                      int aThreadId);
 
   // The circular-ring storage in which this ProfileBuffer stores its data.
-  BlocksRingBuffer mEntries;
+  BlocksRingBuffer& mEntries;
 
  public:
   // `BufferRangeStart()` and `BufferRangeEnd()` return `uint64_t` values